zeek/plugin/include/runtime-support.h

```diff
@@ -529,7 +531,40 @@ inline Val* to_val(const hilti::rt::List<T>& v, BroType* target, std::string_vie
 }
 
 /**
- * Converts a Spicy-side vector to a Zeek value. The result is returned with
+ * Converts a Spicy-side map to a Zeek value. The result is returned with
+ * ref count +1.
+ */
+template<typename K, typename V>
+inline Val* to_val(const hilti::rt::Map<K, V>& m, BroType* target, std::string_view location) {
+    if ( target->Tag() != ::TYPE_TABLE )
+        throw TypeMismatch("map", target, location);
+
+    auto tt = target->AsTableType();
+    if ( tt->IsSet() )
+        throw TypeMismatch("map", target, location);
+
+    auto zv = new TableVal(tt);
+
+    for ( auto i : m ) {
+        if constexpr ( hilti::rt::is_tuple<K>::value )
+            throw TypeMismatch("internal error: sets with tuples not yet supported in to_val()");
```

Comment:
This and the `throw` below leaks `zv` (unless I am missing some ref counting internal to `TableVal`).

One pattern to avoid this could be along these lines:
```
auto zv = std::make_unique<TableVal>(tt); // requires `#include <memory>`

// Do stuff with `zv`.

return zv.release();
```

Reply:
I can move up the throws, but I'll add the make_unique as well (there are more places in the file). The real fix would be converting to Zeek's new intrusivePtr, but that's for later.